### DISCUSSION TITLE
disallow editing part of form by other users

### DISCUSSION THREAD

**Discussion by sghriyadh** [Budibase/budibase#12693] [Help] [answered]
Hi

I have made a form  to ask question and get answer between 4 users A,B,C,D.

Any user can create query or reply to a query. But when any user clicks edit button to answer , he can edit question as well.

How can i disallow editing of data entered by first user(user who created the question)? 

and How can i disable edit button once the answer has been submitted?

**Comment by melohagan** [comment, marked as answer]
Hey @sghriyadh 

> How can i disallow editing of data entered by first user(user who created the question)?

You could add a [Created By](https://docs.budibase.com/docs/users-table#created-by--updated-by-columns) **Auto column** to your question table to indicate who created the question. Then add some [Conditional UI](https://docs.budibase.com/docs/conditions) to make the fields read-only if the `{{ Current User._id }}` matches the question creator.

![Screenshot 2024-01-03 at 14 38 41](https://github.com/Budibase/budibase/assets/101575380/c63f0d92-b098-449d-a9e9-1f3bf0fd9d70)

![Screenshot 2024-01-03 at 14 41 06](https://github.com/Budibase/budibase/assets/101575380/e17a1e16-d805-41a5-bfbb-e1bd2ded4286)

The repeater binding looks like so: `{{ Repeater.Questions.Created By.0._id }}`

---

> How can i disable edit button once the answer has been submitted?

You could also use conditional UI here, and disable the button if the Answer is not empty:

![Screenshot 2024-01-03 at 14 42 44](https://github.com/Budibase/budibase/assets/101575380/9e0df0ff-ae19-4bf3-a47f-b9dd073397d8)

**Reply by sghriyadh** [reply]
i found another solution in the forum.

set a condition , if the field is not empty then it is readonly. 

This way i do not need to depends on useraccounts(i will keep form as public).

Also, once someone has filled in the data they cannot edit it (security,  avoiding any manipulation by any user after they have submitted the data, the fields become uneditable to avoid dispute).
